commit eb9774f0d675922d780f23cbb747f474b0573dea
Author: Avi Kivity <avi@qumranet.com>
Date:   Sun Nov 25 17:45:31 2007 +0200

    KVM: Remove misleading check for mmio during event injection
    
    mmio was already handled in kvm_arch_vcpu_ioctl_run(), so no need to check
    again.
    
    Signed-off-by: Avi Kivity <avi@qumranet.com>

diff --git a/drivers/kvm/x86.c b/drivers/kvm/x86.c
index 2be59364a49d..5e5815604cbd 100644
--- a/drivers/kvm/x86.c
+++ b/drivers/kvm/x86.c
@@ -2344,7 +2344,7 @@ static int __vcpu_run(struct kvm_vcpu *vcpu, struct kvm_run *kvm_run)
 
 	if (irqchip_in_kernel(vcpu->kvm))
 		kvm_x86_ops->inject_pending_irq(vcpu);
-	else if (!vcpu->mmio_read_completed)
+	else
 		kvm_x86_ops->inject_pending_vectors(vcpu, kvm_run);
 
 	vcpu->guest_mode = 1;